python/ray/data/tests/test_dataset_pipeline.py

```diff
@@ -449,6 +491,25 @@ def test_count_sum_on_infinite_pipeline(ray_start_regular_shared):
     assert 9 == pipe.sum()
 
 
+def test_sort_each_window(ray_start_regular_shared):
+    pipe = ray.data.range(12).window(blocks_per_window=3).sort_each_window()
+    assert pipe.take() == list(range(12))
+
+    pipe = (
+        ray.data.range(12)
+        .window(blocks_per_window=12)
+        .sort_each_window(descending=True)
+    )
+    assert pipe.take() == [11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0]
+
```

Comment:
This is the only difference compared to https://github.com/ray-project/ray/pull/26394, rest of change in this PR are the same. To make unit test stable, we'd better only generate 1 window and then sort.

Reply:
Hm, why having blocks_per_window=3 will fail? In this case, it's same as ray.data.range(12).repeat(1) and then sort (i.e. no real windowing).

Reply:
Actuallly after debugging with @clarkzinzow, found out parallelism (and number of blocks) in unit test is changed after https://github.com/ray-project/ray/pull/25883 . Previously the number of blocks / parallelism here is 12, right now is 2, due to in pytest with ray_start_regular_shared cluster fixture, available CPUs is 1, so parallelism becomes 2* CPUs, i.e. 2 here (given data size is very small here). So we should explicitly set number of parallelism here.

Reply:
Ah, I see